controllers/kubeadmconfig_controller.go

```diff
@@ -136,10 +136,12 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 			return ctrl.Result{}, err
 		}
 
-		cloudInitData, err := cloudinit.NewInitControlPlane(&cloudinit.ControlPlaneInput{
+		cpi := &cloudinit.ControlPlaneInput{
 			InitConfiguration:    string(initdata),
 			ClusterConfiguration: string(clusterdata),
-		})
+		}
+		cpi.AdditionalFiles = config.Spec.AdditionalUserDataFiles
```

Comment:
Cannot use promoted fields (promoted from BaseUserData) as struct literal. Hence separate line.

Reply:
You can do it like this:
```go
cloudInitData, err := cloudinit.NewInitControlPlane(&cloudinit.ControlPlaneInput{
  BaseUserData: cloudinit.BaseUserData{
    AdditionalFiles: config.Spec.AdditionalUserDataFiles,
  },
  InitConfiguration: string(initdata),
  // ...
}
```

Reply:
This is an embedded type, from the second paragraph in this section https://golang.org/ref/spec#Struct_types:

> The unqualified type name acts as the field name.


You've just got to use the field name like this:

```
		joinData, err := cloudinit.NewJoinControlPlane(&cloudinit.ControlPlaneJoinInput{
			// TODO do a len check or something here
			ControlPlaneAddress: fmt.Sprintf("https://%s:%d", cluster.Status.APIEndpoints[0].Host, cluster.Status.APIEndpoints[0].Port),
			JoinConfiguration:   string(joinBytes),
			BaseUserData: cloudinit.BaseUserData{
				AdditionalFiles: config.Spec.AdditionalUserDataFiles,
			},
		})
```

Reply:
oops. I really need to refresh my tabs more frequently. Apologies for the duplication @ncdc

Reply:
It's all good!

Reply:
Thank you for the suggestion @ncdc  and @chuckha 😄 . PR updated.